[move] Initial doc migration for 2024 (#16597)
MystenLabs/sui: external-crates/move/documentation/book/src/modules.md

[package settings](./packages.md).

Modules should normally be declared using [named addresses](./address.md) (as opposed to using the
numerical value directly). For example:

```move
module test_addr::test {
    public struct Example has copy, drop { a: address }

    friend test_addr::another_test;

    public fun print() {
        let example = Example { a: @test_addr };
        debug::print(&example)
    }
}
```

These named addresses commonly match the name of the [package](./packages.md).

Because named addresses only exist at the source language level and during compilation, named
addresses will be fully substituted for their value at the bytecode level. For example if we had the
following code:

```move
fun example() {
    my_addr::m::foo(@my_addr);
}
```

and we compiled it with `my_addr` set to `0xC0FFEE`, then it would be equivalent to the following
operationally:

```move
fun example() {
    0xC0FFEE::m::foo(@0xC0FFEE);
}
```

While at the source level these two different accesses are equivalent, it is a best practice to
always use the named address and not the numerical value assigned to that address.

> Module names can start with letters `a` to `z` or letters `A` to `Z`. After the first character,

Reviewer: ```suggestion
Module names can start with a lowercase letter from `a` to `z` or an uppercase letter from `A` to `Z`. After the first character,
```
Adding suggested words might help clear any confusion of whether the repeat is intentional. Wouldn't be needed in next line.

module names can contain underscores `_`, letters `a` to `z`, letters `A` to `Z`, or digits `0` to
`9`.

```move
module a::my_module {}
module a::foo_bar_42 {}
```

> Typically, module names start with an lowercase letter. A module named `my_module` should be stored

Reviewer: ```suggestion
Typically, module names start with a lowercase letter. A module named `my_module` should be stored
```

in a source file named `my_module.move`.

## Members

All members inside a `module` block can appear in any order. Fundamentally, a module is a collection

> of [`types`](./structs.md) and [`functions`](./functions.md). The [`use`](./uses.md) keyword is used

Reviewer: ```suggestion
of [`types`](./structs.md) and [`functions`](./functions.md). The [`use`](./uses.md) keyword
```

> to refer to members from other modules. The [`const`](./constants.md) keyword defines constants that

Reviewer: ```suggestion
refers to members from other modules. The [`const`](./constants.md) keyword defines constants that
```

can be used in the functions of a module.

> [`friend`](./friends.md) are deprecated concept for specifying a list of trusted modules. The

Reviewer: ```suggestion
The [`friend`](./friends.md) syntax is a deprecated concept for specifying a list of trusted modules. The
```
